# HG changeset patch
# User Kai Habel <kai.habel@gmx.de>
# Date 1312136109 -7200
#      Sun Jul 31 20:15:09 2011 +0200
# Node ID 35ea8b9b2a8ef5e3bff7517bd8f33c27962c2c04
# Parent  b9c66f8912819d632f96c6f90ab2646ed93fe908
__init_fltk__.cc: show canvas only, when figure is visible (bug #33321)

diff --git a/src/DLD-FUNCTIONS/__init_fltk__.cc b/src/DLD-FUNCTIONS/__init_fltk__.cc
--- a/src/DLD-FUNCTIONS/__init_fltk__.cc
+++ b/src/DLD-FUNCTIONS/__init_fltk__.cc
@@ -856,18 +856,21 @@ public:
             hide_menubar ();
 
           mark_modified();
       }
   }
 
   void show_canvas (void)
   {
-    canvas->show ();
-    canvas->make_current ();
+    if (fp.is_visible ())
+      {
+        canvas->show ();
+        canvas->make_current ();
+      }
   }
 
   void hide_canvas (void)
   {
     canvas->hide ();
   }
 
   void mark_modified (void)
